source/Calamari.Aws/Kubernetes/Discovery/AwsKubernetesDiscoverer.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Amazon;
+using Amazon.EKS;
+using Amazon.EKS.Model;
+using Calamari.Common.Features.Discovery;
+using Calamari.Common.Plumbing.Logging;
+using Calamari.Common.Plumbing.Variables;
+using Calamari.Kubernetes.Aws;
+using Newtonsoft.Json;
+using Octopus.CoreUtilities.Extensions;
+
+namespace Calamari.Aws.Kubernetes.Discovery
+{
+    public class AwsKubernetesDiscoverer : IKubernetesDiscoverer
+    {
+        readonly ILog log;
+
+        public AwsKubernetesDiscoverer(ILog log)
+        {
+            this.log = log;
+        }
+
+        public string Name => "Aws";
+        
+        public IEnumerable<KubernetesCluster> DiscoverClusters(string contextJson, IVariables variables)
+        {
+            if (!TryGetAuthenticationDetails(contextJson, out var authenticationDetails))
+                yield break;
```

Comment:
I've updated it to log a warning, is that ok? I've also updated the Azure discoverer to log a warning in the same spot as I missed it for that one too.

Reply:
Our philosophy here is that target discovery errors should not prevent a deployment from proceeding, so from that point of view logging a warning would be appropriate.

It would be nice to actually get an error in the server logs if we think think we have detected a code defect, but since this is in Calamari, that is not trivial to achieve, and not worth investing in just for this unlikely scenario.